lib/dom_utils.coffee

```diff
@@ -315,6 +315,25 @@ DomUtils =
         false
     handlerStack.suppressEvent
 
+  consumeKeyup: (event, callback = null) ->
+    @suppressEvent event
+    keyChar = KeyboardUtils.getKeyCharString event
+    handlerStack.push
+      _name: "dom_utils/consumeKeyup"
+      keydown: (event) ->
+        @remove()
+        handlerStack.continueBubbling
+      keyup: (event) ->
+        return handlerStack.continueBubbling unless keyChar == KeyboardUtils.getKeyCharString event
+        @remove()
+        handlerStack.suppressEvent
+      # We cannot track keyup events if we lose the focus.
+      blur: (event) ->
+        @remove() if event.target == window
+        handlerStack.continueBubbling
+    callback?()
+    handlerStack.suppressEvent
+
```

Comment:
On further reflection, this code (or any alternative) should use `event.code`, not `event.key`. Otherwise, the `keyup` may have a different `key` (eg. `keydownEvent.key == "F"`, `keyupEvent.key == "f"`) if a modifier is released before the key itself, whereas we only care about the physical key, as described by `code`.